qiskit/transpiler/passmanager.py

```diff
@@ -312,3 +312,179 @@ def passes(self) -> List[Dict[str, BasePass]]:
                 item["flow_controllers"] = {}
             ret.append(item)
         return ret
+
+
+class FullPassManager(PassManager):
+    """A full Pass manager pipeline for a backend
```

Comment:
I'm not set on the name, I'll use one of those instead. I went with `FullPassManager` to represent a **full** transpilation pipeline. But I agree it could be confusing.